src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
+        this.cache = new ConcurrentHashMap<>();
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> fillAndReturnUserData(userId), executor);
+    }
+
+    private UserData fillAndReturnUserData(String userId) {
+        try {
+            Thread.sleep(200);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        if (cache.containsKey(userId)) {
+            return cache.get(userId);
```

Comment:
Accessing a `ConcurrentHashMap` with `containsKey` followed by `get` can be inefficient because it performs two separate operations and the map may change between calls. It is better to use `computeIfAbsent` which is atomic and performs the check and retrieval in a single step.